dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/configurations/ApiClients.kt

```diff
@@ -0,0 +1,37 @@
+package org.dataland.datalandcommunitymanager.configurations
+
+import okhttp3.OkHttpClient
+import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
+import org.dataland.datalandbackend.openApiClient.api.MetaDataControllerApi
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+/**
+ * A configuration class that provides access to pre-configured Api Clients
+ */
+@Configuration
+class ApiClients(
+    @Value("\${dataland.backend.base-url}") private val backendBaseUrl: String,
+) {
+    /**
+     * Creates an auto-authenticated version of the CompanyDataControllerApi of the backend
+     */
+    @Bean
+    fun getCompanyDataControllerApi(
+        @Qualifier("AuthenticatedOkHttpClient") authenticatedOkHttpClient: OkHttpClient,
+    ): CompanyDataControllerApi {
+        return CompanyDataControllerApi(backendBaseUrl, authenticatedOkHttpClient)
+    }
+
+    /**
+     * Creates an auto-authenticated version of the MetaDataControllerApi of the backend
+     */
+    @Bean
+    fun getMetaDataControllerApi(
+        @Qualifier("AuthenticatedOkHttpClient") authenticatedOkHttpClient: OkHttpClient,
+    ): MetaDataControllerApi {
+        return MetaDataControllerApi(backendBaseUrl, authenticatedOkHttpClient)
+    }
+}
```

Comment:
What is this new file needed for, i.e. where are the functions `getCompanyDataControllerApi` and `getMetaDataControllerApi` called?

Reply:
Methods annotated with @Bean are providers for objects that get autowired into components' (@Component) or services' (@Service) constructor which are annotated with @Autowired. So I do not call these functions but Spring does internally for dependency injection.